modules/core/04-channel/v2/keeper/msg_server.go

```diff
@@ -47,8 +48,34 @@ func (*Keeper) Acknowledgement(ctx context.Context, acknowledgement *channeltype
 }
 
 // RecvPacket implements the PacketMsgServer RecvPacket method.
-func (*Keeper) RecvPacket(ctx context.Context, packet *channeltypesv2.MsgRecvPacket) (*channeltypesv2.MsgRecvPacketResponse, error) {
-	panic("implement me")
+func (k *Keeper) RecvPacket(ctx context.Context, msg *channeltypesv2.MsgRecvPacket) (*channeltypesv2.MsgRecvPacketResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	packet := msg.Packet
+	err := k.recvPacket(ctx, packet, msg.ProofCommitment, msg.ProofHeight)
+	if err != nil {
+		sdkCtx.Logger().Error("send packet failed", "source-id", packet.SourceId, "error", errorsmod.Wrap(err, "send packet failed"))
+		return nil, errorsmod.Wrapf(err, "send packet failed for source id: %s", packet.SourceId)
```

Comment:
we should be logging receive packet here, we can also include the dest-id maybe here